sky/serve/load_balancer.py

```diff
@@ -77,23 +78,85 @@ def _sync_with_controller(self):
                         ready_replica_urls)
             time.sleep(constants.LB_CONTROLLER_SYNC_INTERVAL_SECONDS)
 
-    async def _redirect_handler(self, request: fastapi.Request):
-        self._request_aggregator.add(request)
-        ready_replica_url = self._load_balancing_policy.select_replica(request)
-
-        if ready_replica_url is None:
-            raise fastapi.HTTPException(status_code=503,
-                                        detail='No ready replicas. '
-                                        'Use "sky serve status [SERVICE_NAME]" '
-                                        'to check the replica status.')
+    async def _proxy_request_to(
+            self, url: str,
+            request: fastapi.Request) -> Optional[fastapi.responses.Response]:
+        """Proxy the request to the specified URL.
 
-        path = f'http://{ready_replica_url}{request.url.path}'
-        logger.info(f'Redirecting request to {path}')
-        return fastapi.responses.RedirectResponse(url=path)
+        Returns:
+            The response from the endpoint replica. None if anything goes wrong.
```

Comment:
We can add a `Yields` field for the stuffs we are yielding.

Reply:
This function seems not yielding anything. Added a docstr for the `stream_response` function.